sematic/scheduling/job_details.py

```diff
@@ -0,0 +1,234 @@
+            "This is meant to emulate an Enum and not to be instantiated directly"
+        )
+
+    Requested = "Requested"
+    Pending = "Pending"
+    Running = "Running"
+    Restarting = "Restarting"
+    Succeeded = "Succeeded"
+    Failed = "Failed"
+    Deleted = "Deleted"
+
+    @classmethod
+    def is_active(cls, state) -> bool:
+        return state in _ACTIVE_STATES
+
+
+_ACTIVE_STATES = {
+    KubernetesJobState.Requested,
+    KubernetesJobState.Pending,
+    KubernetesJobState.Running,
+    KubernetesJobState.Restarting,
+}
+
+KubernetesJobStateString = Literal[
+    "Requested",
+    "Pending",
+    "Running",
+    "Restarting",
+    "Succeeded",
+    "Failed",
+    "Deleted",
+]
+
+
+class JobKind:
+    def __init__(self) -> None:
+        raise RuntimeError("This is meant to emulate an enum, not be instantiated.")
+
+    resolver = "resolver"
+    run = "run"
+
+
+JobKindString = Literal[
+    "resolver",
+    "run",
+]
+
+
+@dataclass
+class PodSummary:
+    """Summary of an individual pod associated with a job."""
+
+    # Unique name of the pod assigned by K8s
+    pod_name: str
+
+    # number of times the container within the pod has
+    # been restarted. Is NOT the number of pods that
+    # has been associated with the associated job.
+    container_restart_count: Optional[int] = None
+
+    # Current phase in the lifecycle of the pod.
+    phase: Optional[str] = None
+
+    # Human readable message associated with the most recent
+    # "relevant" condition of the job. Relevancy is based on
+    # condition kind and True/False value.
+    condition_message: Optional[str] = None
+
+    # Human readable reason why the pod is unschedulable
+    # (assuming the pod is currently unschedulable). Note that
+    # just because a pod is *currently* unschedulable doesn't mean
+    # that autoscaling or workload drops won't make it schedulable at
+    # some point.
+    unschedulable_message: Optional[str] = None
+
+    # Human readable message about the state of the container
+    # associated with the pod.
+    container_condition_message: Optional[str] = None
+
+    # Exit code of the container, assuming the container has exited and
+    # the exit code can be identified.
+    container_exit_code: Optional[int] = None
+
+    # Epoch time that Kubernetes started the pod.
+    start_time_epoch_seconds: Optional[float] = None
+
+    # Name of the node the pod is running on, if it
+    # has been scheduled to a node.
+    node_name: Optional[str] = None
+
+    # Indicator of whether the pod has shown some abnormality
+    # that shows that Sematic should move the current run to
+    # a terminal state (or retry it).
+    detected_infra_failure: bool = False
+
+    def string_summary(self) -> str:
+        return (
+            f"{self.pod_name}[in phase '{self.phase}']"
+            f"[{self.condition_message}]"
+            f"[{self.container_condition_message}]"
+        )
+
+
+@dataclass(frozen=True)
+class JobStatus:
```

Comment:
I believe this can be kept generic, without tying it to Kubernetes, as it's fair to assume all platforms can provide the 3 fields. It can be instantiated with K8 values.